### DISCUSSION TITLE
Is this situation normal?

### DISCUSSION THREAD

**Discussion by disburden** [prisma/prisma#21908] [Q&A] [answered]
### Question

I have an "Order" table, and he has a one-to-one relationship with the "PayDetail" table, that is, the order will have an optional payment details record.
This is the model that I have created.
```prisma
model Order {
  orderId   Int        @id @default(autoincrement())
  product   String
  count     Int
  money     Decimal    @db.Decimal(9, 2)
  payDetail PayDetail?
}

model PayDetail {
  id         Int      @id @default(autoincrement())
  createTime DateTime @default(now())
  detail     Json
  order      Order?   @relation(fields: [aOrderId], references: [orderId])
  aOrderId   Int?     @unique
}
```

When I executed the "db push" command, I found that there is no "PayDetail" field in the database, but only the "aOrderId" associated field in the PayDetail schedule.
![微信图片_20231112170357](https://github.com/prisma/prisma/assets/1391188/e233d7c6-f445-44f8-91ca-d7e32a12f3cc)

Is this normal?if this situation is not normal, how should I design this relationship?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @disburden 👋 

Yes, this is normal and expected behavior. In Prisma, the [relation fields](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#relation-fields) like `payDetail` in your `Order` model and `order` in your PayDetail model are virtual and used to establish relations between models. They do not correspond to actual fields in the database.
The actual connection between the `Order` and `PayDetail` tables in the database is represented by the `aOrderId` field in the `PayDetail` model. This field is a foreign key that references the `orderId` in the `Order` table.
Here is the relevant part of your schema:

```prisma
model Order {
  orderId   Int        @id @default(autoincrement())
  product   String
  count     Int
  money     Decimal    @db.Decimal(9, 2)
  payDetail PayDetail?
}

model PayDetail {
  id         Int      @id @default(autoincrement())
  createTime DateTime @default(now())
  detail     Json
  order      Order?   @relation(fields: [aOrderId], references: [orderId])
  aOrderId   Int?     @unique
}
```

In this schema, the `payDetail` field in the `Order` model is optional (PayDetail?), which means an order can have zero or one payment detail. On the other side, the `order` field in the `PayDetail` model is also optional (Order?), which means a payment detail can be associated with zero or one order. The @unique attribute on the `aOrderId` field ensures that each `PayDetail` can be associated with at most one `Order`.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by disburden** [comment]
Ok, this situation can meet my business needs, I just want to make sure that this situation is not abnormal.
tks!
